x-pack/plugins/security_solution/server/endpoint/routes/trusted_apps/index.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from 'kibana/server';
+import { GetTrustedAppsRequestSchema } from '../../../../common/endpoint/schema/trusted_apps';
+import { TRUSTED_APPS_LIST_API } from '../../../../common/endpoint/constants';
+import { getTrustedAppsListRouteHandler } from './handlers';
+import { EndpointAppContext } from '../../types';
+
+export const registerTrustedAppsRoutes = (
+  router: IRouter,
+  endpointAppContext: EndpointAppContext
+) => {
+  // GET list
+  router.get(
+    {
+      path: TRUSTED_APPS_LIST_API,
+      validate: GetTrustedAppsRequestSchema,
```

Comment:
I noticed that all other schemas are just made inline instead of separate files. It seems to me to make sense - controller layer should only contain HTTP mapping related functionality hence it is already separate responsibility (separate concern) and splitting it further only creates more indirection, rather keeping path, schema and other things inline here gives a really good overview of API.

Reply:
I see multiple patterns  across the app and Kibana. Defining it inline here makes it harder (impossible?) to test because currently we are not actually simulating an HTTP request to the Kibana server (not sure thats supported in the UT framework - ex. does kibana core provide a mock "server"?), but rather we test against the HTTP handler.

Also - as I learned from Frank, its best to ensure we test our schemas separately so that if we do end up changing schema libraries (ex. move to `io-ts`), we can ensure that our expectation remain met with the new lib.

Reply:
One thing I did like about the everything-defined-here pattern was that the schema was close to the URL route. This adds two layers of separation. Need to look in constants to see what the route is, and handlers to see what the params are.